### ISSUE TITLE
[Design] Should operator methods be definable on types that are not involved?

### ISSUE THREAD

**Issue by lydia-duncan** [chapel-lang/chapel#17514]
At first effort in the operator methods implementation, we've allowed operator methods to be defined on a type that isn't used by any of the arguments, e.g.

```chapel
operator T1.+(lhs: T2, rhs: T3) { ... }
```

However, this meant that I was able to accidentally associate an operator method with the wrong type without noticing.  Engin points out that there's not necessarily much benefit in providing it, and I'm realizing it could lead to surprises if a user imports a type and suddenly an operator between two different types starts behaving differently.  Should we prevent this from being possible?

Note that while this is a breaking change (in that it disables something that was available), I'm not as concerned about breaking it since operator methods are a new feature and are not expected to be production stable (at least as of the 1.24 release)

**Comment by bradcray** [comment]
I agree that this behavior is weird, but I'm not sure I'd outlaw it proactively.  It seems like there may some non-obvious relationship to the type that might cause one to want to do this? (a subclass or parent class being one case; T2 or T3 having fields of type T1 being another potential one; maybe wanting to create a type that acts module-like as a third one?).  I feel like there's plenty about operator overloading in general that is subject to abuse such that this feels minor/unlikely relative to other cases.  I'd be open to putting in a warning for such cases (but think of it as low priority).
